week2_shoppinglist/index.js

```diff
@@ -0,0 +1,384 @@
+    liked: false,
+  },
+  {
+    id: 2,
+    category: "stationery",
+    image: "./images/bag.jpg",
+    name: "잔망루피 쇼핑백 (2종/택1)",
+    price: 1500,
+    liked: false,
+  },
+  {
+    id: 3,
+    category: "stationery",
+    image: "./images/sticker.jpg",
+    name: "잔망루피 스티커",
+    price: 1500,
+    liked: false,
+  },
+  {
+    id: 4,
+    category: "fashion",
+    image: "./images/doll.jpg",
+    name: "잔망루피 파자마 봉제 인형",
+    price: 29000,
+    liked: false,
+  },
+  {
+    id: 5,
+    category: "fashion",
+    image: "./images/passport.jpg",
+    name: "잔망루피 여권 케이스",
+    price: 5900,
+    liked: false,
+  },
+  {
+    id: 6,
+    category: "fashion",
+    image: "./images/flower.jpg",
+    name: "잔망루피 꽃다발 봉제 인형",
+    price: 34000,
+    liked: false,
+  },
+  {
+    id: 7,
+    category: "digital",
+    image: "./images/mouse.jpg",
+    name: "잔망루피 마우스패드",
+    price: 4000,
+    liked: false,
+  },
+  {
+    id: 8,
+    category: "digital",
+    image: "./images/usb.jpg",
+    name: "로이체 잔망루피 5in1 멀티 USB 허브",
+    price: 36900,
+    liked: false,
+  },
+  {
+    id: 9,
+    category: "digital",
+    image: "./images/mic.jpg",
+    name: "로이체 잔망루피 블루투스 마이크 스피커",
+    price: 39900,
+    liked: false,
+  },
+];
+
+// 상품 리스트 렌더링
+function renderItems(filterCategory = "all") {
+  const filteredItems = SHOPPING_LIST.filter((item) => {
+    return filterCategory === "all" || item.category === filterCategory; // 상품 카테고리 필터링
+  });
+
+  const itemsContainer = document.querySelector(".items"); // 상품 리스트를 렌더링할 class 가져오기
+  itemsContainer.innerHTML = ""; // 기존 상품 목록 비우기
+
+  // 상품 목록 추가
+  filteredItems.forEach((item) => {
+    const article = document.createElement("article"); // 각 상품에 대한 article 생성
+    article.className = "item";
+    article.onclick = function () {
+      addCart(item);
+    };
+
+    // 이미지
+    const img = document.createElement("img");
+    img.src = item.image;
+    img.alt = "상품 이미지";
+    img.className = "item-image";
+
+    // 좋아요
+    const likeDiv = document.createElement("div");
+    likeDiv.className = "like";
+
+    // 상품명
+    const nameP = document.createElement("p");
+    nameP.className = "name";
+    nameP.textContent = item.name;
+
+    // 가격
+    const priceP = document.createElement("p");
+    priceP.className = "price";
+    priceP.textContent =
+      item.price.toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ",") +
+      "원"; // 천단위 표시
+
+    // 요소 추가
+    article.appendChild(img);
+    article.appendChild(likeDiv);
+    article.appendChild(nameP);
+    article.appendChild(priceP);
+    itemsContainer.appendChild(article);
+  });
+}
+
+//장바구니 목록 추가
+function addCart(item) {
+  let cartList = JSON.parse(localStorage.getItem("cartList")) || []; // cartList가 배열이 아닌 경우, 빈 배열로 초기화
+  cartList.push(item); // 새 상품 추가
+  localStorage.setItem("cartList", JSON.stringify(cartList)); // 변경된 장바구니 목록을 로컬 스토리지에 저장
+  alert(`${item.name}을(를) 장바구니에 담았습니다.`);
+  var answer = confirm("장바구니로 이동하시겠습니까?");
```

Comment:
p3: var는 웬만하면 안 쓰는 게 좋아! 여기서는 const로 바꾸는 건 어떨까!?
```suggestion
  const answer = confirm("장바구니로 이동하시겠습니까?");
```

Reply:
var, let, const 차이점도 같이 살펴보면 좋을 거같아염ㅎㅎ

https://velog.io/@bathingape/JavaScript-var-let-const-%EC%B0%A8%EC%9D%B4%EC%A0%90

Reply:
오오...! 벨로그까지!! 제안 감사합니다! 수정했습니다!